app/controllers/api/v1/videos_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Api::V1::VideosController < ApplicationController
+  before_action :check_admin, only: [:create, :update, :destroy]
+
+  def index
+    videos = Video.where(deleted_at: nil)
+    render json: videos, status: 200
+  end
+
+  def show
+    video = Video.find(params[:id])
+    render json: video, status: 200
+  end
+
+  def create
+    video = current_user.mcs_admin.videos.build(video_params)
+    if video.save
```

Comment:
I feel like its fine how it is. Mainly because if we do that then all of our controllers will have that method when we might not necessarily want that.